metrics/sink.go

```diff
@@ -15,143 +15,186 @@ var (
 	_ Sink = &DummySink{}
 )
 
+// Sink collects Sample's values.
 type Sink interface {
-	Add(s Sample)                              // Add a sample to the sink.
-	Format(t time.Duration) map[string]float64 // Data for thresholds.
-	IsEmpty() bool                             // Check if the Sink is empty.
+	// Add adds a sample to the sink.
+	Add(s Sample)
+
+	// IsEmpty checks if the Sink has never received a value.
+	IsEmpty() bool
 }
 
 type CounterSink struct {
-	Value float64
-	First time.Time
+	value float64
+	first time.Time
 }
 
 func (c *CounterSink) Add(s Sample) {
-	c.Value += s.Value
-	if c.First.IsZero() {
-		c.First = s.Time
+	c.value += s.Value
+	if c.first.IsZero() {
+		c.first = s.Time
 	}
 }
 
+// LastValue returns the counter value.
+func (c *CounterSink) LastValue() float64 {
```

Comment:
This shouldn't be called `LastValue`, that only makes sense for the `Gauge` metrics. Either `Value()` or `Total()` makes more sense.